python-client/giskard/core/model_validation.py

```diff
@@ -86,7 +86,14 @@ def validate_model_execution(model: BaseModel, dataset: Dataset, deterministic:
     try:
         prediction = model.predict(validation_ds)
     except Exception as e:
-        raise ValueError(error_message) from e
+        if "index 1 is out of bounds for axis 0 with size 1" in str(e) and len(validation_ds.df.columns) == 1:
```

Comment:
I think it's not a good idea to depends on the message.

It would be better to try to handle it using the exception type (IndexError I think here)

Reply:
```python
   prediction = None
   try:
        prediction = model.predict(validation_ds)
    except IndexError:
       model.accepts_only_pd_series = True
    except Exception as e:
       raise ValueError(error_message) from e
    if prediction is None:
        try:
           prediction = model.predict(validation_ds)
        except Exception as e:
           raise ValueError(error_message) from e
```

Reply:
I wanted to capture not any `IndexError` but specifically this one `"index 1 is out of bounds for axis 0 with size 1"`

Reply:
Hmm, ok I see. I think you should still match on exception type also, in addition to checking message.

Small detail, but you can avoid the else by inversing your if 
```python
if "index 1 is out of bounds for axis 0 with size 1" not in str(e) or number_of_features != 1:
    raise ValueError(error_message) from e

try:
    model.accepts_only_pd_series = True
    prediction = model.predict(validation_ds)
except Exception as e2:
    raise ValueError(error_message) from e2

````

Reply:
Thanks! done ✅